### PR TITLE
Fix issue 663 project nodes always yellow and 412 dependencies nodes visibly slow to show

### LINKED ISSUE #663: In most cases DesignTime build returns empty ResolvedProjectReferences data
Very rarely i saw receiving expected data in snapshots for ResolvedProjectReferences , in most cases when your projecA references project B we receive empty snapshots for ResolvedProjectRefernces. Could be something with ResolveProjectReferencesDesignTime  target or some weird timing in CPS, if they have any specific logic for ResolvedProjectReference rules.

### PR DESCRIPTION
**Pull request by abpiskunov**
In progress: 
- still adding unit tests for changes

@dotnet/project-system 

Fixes #663 and #412

### CODE REVIEW

File: src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/ProjectDependenciesSubTreeProvider.cs [hunk 1]
```diff
@@ -90,6 +90,22 @@ protected override IDependencyNode CreateRootNode()
...
+            }
+
+            var projectNode = RootNode.Children.FirstOrDefault(x => x.Id.ToNormalizedId().Equals(nodeId));
+            if (projectNode != null)
+            {
+                projectNode.AddChild(projectNode);
+            }
+
+            return projectNode;
+        }
```
Comment (abpiskunov) [inline src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Tree/Dependencies/ProjectDependenciesSubTreeProvider.cs]: This is not needed for this change, and was part of another change i am working on. Will revert it when add unit tests in next commit

**Comment by abpiskunov** [comment]
also added fixes for #431 and #635 , tests are added. Will merge after build is green and follow up with feedback later if any.

**Comment by srivatsn** [comment]
Hey Anton, we are in ask mode and all fixes need to be reviewed and approved before merging.

**Comment by srivatsn** [comment]
And regardless of ask mode, non-trivial PRs need to be reviewed before they are merged in. Please revert and re-create the PR.

**Pull request merged**